joern-cli/frontends/swiftsrc2cpg/src/test/scala/io/joern/swiftsrc2cpg/passes/ast/CallTests.scala

```diff
@@ -111,6 +113,148 @@ class CallTests extends SwiftCompilerSrc2CpgSuite {
       returnId.typeFullName shouldBe "SwiftTest.Foo"
     }
 
+    "be correct for simple calls to functions from extensions" in {
+      val testCode =
+        """
+          |extension Foo {
+          |  func foo() {}
+          |  func bar() {}
+          |}
+          |class Foo {
+          |  func main() {
+          |    foo()
+          |    self.bar()
+          |  }
+          |}
+          |""".stripMargin
+      val cpg = code(testCode)
+
+      val List(fooCall) = cpg.call.nameExact("foo").l
+      fooCall.methodFullName shouldBe x2cpg.Defines.DynamicCallUnknownFullName
+      fooCall.signature shouldBe ""
+      fooCall.dispatchType shouldBe DispatchTypes.DYNAMIC_DISPATCH
+
+      val List(fooCallReceiver) = fooCall.receiver.isIdentifier.l
+      fooCallReceiver.name shouldBe "self"
+      fooCallReceiver.typeFullName shouldBe "Sources/main.swift:<global>.Foo"
+
+      val List(barCall) = cpg.call.nameExact("bar").l
+      barCall.methodFullName shouldBe x2cpg.Defines.DynamicCallUnknownFullName
+      barCall.signature shouldBe ""
+      barCall.dispatchType shouldBe DispatchTypes.DYNAMIC_DISPATCH
+
+      val List(barCallReceiverCall) = barCall.receiver.isIdentifier.l
+      barCallReceiverCall.name shouldBe "self"
+      barCallReceiverCall.typeFullName shouldBe "Sources/main.swift:<global>.Foo"
+    }
+
+    "be correct for simple calls to functions from extensions with compiler support" in {
+      val testCode =
+        """
+          |extension Foo {
+          |  func foo() {}
+          |  func bar() {}
+          |}
+          |class Foo {
+          |  func main() {
+          |    foo()
+          |    self.bar()
+          |  }
+          |}
+          |""".stripMargin
+      val cpg = codeWithSwiftSetup(testCode)
+
+      val List(fooCall) = cpg.call.nameExact("foo").l
+      fooCall.methodFullName shouldBe "SwiftTest.Foo.foo:()->()"
+      fooCall.signature shouldBe "()->()"
+      fooCall.dispatchType shouldBe DispatchTypes.DYNAMIC_DISPATCH
+      val List(fooCallReceiver) = fooCall.receiver.isIdentifier.l
+      fooCallReceiver.name shouldBe "self"
+      fooCallReceiver.typeFullName shouldBe "SwiftTest.Foo"
+
+      val List(barCall) = cpg.call.nameExact("bar").l
+      barCall.methodFullName shouldBe "SwiftTest.Foo.bar:()->()"
+      barCall.signature shouldBe "()->()"
+      barCall.dispatchType shouldBe DispatchTypes.DYNAMIC_DISPATCH
+
+      val List(barCallReceiverCall) = barCall.receiver.isIdentifier.l
+      barCallReceiverCall.name shouldBe "self"
+      barCallReceiverCall.typeFullName shouldBe "SwiftTest.Foo"
```

Comment:
Extension methods are statically dispatched in Swift.

Reply:
We are talking about Extensions (https://docs.swift.org/swift-book/documentation/the-swift-programming-language/extensions/) here. Are you sure?
E.g.,
```
extension Int {
    mutating func square() {
        self = self * self
    }
}
var someInt = 3
someInt.square()
```

Reply:
That was just an example, did not pick it on purpose.

Ok, then we have to model these extension calls as static calls. Currently there is no way to detect this as the swift-parser AST does not carry this information. I need to extend the compiler info queries to be able to detect static calls reliably.
This is out of the scope here (this PR is already large). I will change these tests to `pendingUntilFixed`.